ployst/github/hooker.py

```diff
@@ -0,0 +1,87 @@
+from datetime import datetime
+import hashlib
+import hmac
+import json
+
+
+DEBUG = True
+
+
+class GithubHookException(Exception):
+    pass
+
+
+def compute_github_signature(body, secret):
+    return hmac.new(secret, body, hashlib.sha1).hexdigest()
+
+
+class GithubHookHandler(object):
+    """
+    A general purpose-class to help build github hook handlers.
+
+    See https://developer.github.com/webhooks/ for Github's webhook spec.
+
+    NOTE: This module may be spin-off into a separate library.
+    """
+
+    @staticmethod
+    def get_org_repo(request):
```

Comment:
Yes the issue is that we need to parse the payload first and check whether the request is legit afterwards... meaning we can't do both cleanly without duplication in this class, that I want to publish as a separate library. Maybe the generation of a secret per organisation could be part of this library as well, and be *standard behaviour*?